test/widgets/text_test.dart

```diff
@@ -7,6 +7,63 @@ import 'package:zulip/widgets/text.dart';
 import '../flutter_checks.dart';
 
 void main() {
+  group('zulipTypography', () {
+    Future<Typography> getZulipTypography(WidgetTester tester, {
+      required bool platformRequestsBold,
+    }) async {
+      late final Typography result;
+      await tester.pumpWidget(
+        MediaQuery(data: MediaQueryData(boldText: platformRequestsBold),
+          child: Builder(builder: (context) {
+            result = zulipTypography(context);
+            return const SizedBox.shrink();
+          })));
+      return result;
+    }
+
+    final matchesFonts = it<TextStyle>()
+      ..fontFamily.equals(kDefaultFontFamily)
+      ..fontFamilyFallback.isNotNull().deepEquals(defaultFontFamilyFallback);
```

Comment:
nit:
```suggestion
    final matchesFontFamilies = it<TextStyle>()
      ..fontFamily.equals(kDefaultFontFamily)
      ..fontFamilyFallback.isNotNull().deepEquals(defaultFontFamilyFallback);
```

since it's not concerned with the font weight, etc., only the font families